<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>BallGame Level Editor</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout_2">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="LevelGraphicsView" name="graphicsView"/>
      </item>
      <item>
       <widget class="QTabWidget" name="tabWidget">
        <property name="minimumSize">
         <size>
          <width>210</width>
          <height>0</height>
         </size>
        </property>
        <property name="currentIndex">
         <number>1</number>
        </property>
        <widget class="QWidget" name="tab">
         <attribute name="title">
          <string>Objects</string>
         </attribute>
         <widget class="QComboBox" name="objectSelectorComboBox">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>10</y>
            <width>251</width>
            <height>22</height>
           </rect>
          </property>
         </widget>
         <widget class="QWidget" name="layoutWidget">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>40</y>
            <width>258</width>
            <height>455</height>
           </rect>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <widget class="QTableWidget" name="objectsTableWidget">
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>290</height>
              </size>
             </property>
             <column>
              <property name="text">
               <string>Key</string>
              </property>
             </column>
             <column>
              <property name="text">
               <string>Value</string>
              </property>
             </column>
            </widget>
           </item>
           <item>
            <layout class="QVBoxLayout" name="verticalLayout_2">
             <item>
              <widget class="QPushButton" name="pushButton_4">
               <property name="text">
                <string>Add Property</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="pushButton_5">
               <property name="text">
                <string>Delete Property</string>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="verticalSpacer">
               <property name="orientation">
                <enum>Qt::Vertical</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>20</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item>
              <widget class="QPushButton" name="pushButton_3">
               <property name="text">
                <string>Copy Object</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="pushButton">
               <property name="text">
                <string>Add Blank New Object</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="pushButton_2">
               <property name="text">
                <string>Delete Object</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </widget>
        <widget class="QWidget" name="tab_2">
         <attribute name="title">
          <string>Properties</string>
         </attribute>
         <widget class="QTableWidget" name="levelPlistTableWidget">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>0</y>
            <width>256</width>
            <height>431</height>
           </rect>
          </property>
          <column>
           <property name="text">
            <string>Key</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Value</string>
           </property>
          </column>
         </widget>
         <widget class="QPushButton" name="pushButton_6">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>440</y>
            <width>261</width>
            <height>23</height>
           </rect>
          </property>
          <property name="text">
           <string>Add Property</string>
          </property>
         </widget>
         <widget class="QPushButton" name="pushButton_7">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>470</y>
            <width>261</width>
            <height>23</height>
           </rect>
          </property>
          <property name="text">
           <string>Delete Property</string>
          </property>
         </widget>
        </widget>
       </widget>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout"/>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionNew_Level"/>
    <addaction name="separator"/>
    <addaction name="actionOpen_Level"/>
    <addaction name="separator"/>
    <addaction name="actionSave"/>
    <addaction name="actionSave_As"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionNew_Level">
   <property name="text">
    <string>New</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionOpen_Level">
   <property name="text">
    <string>Open</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionSave_As">
   <property name="text">
    <string>Save As</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>LevelGraphicsView</class>
   <extends>QGraphicsView</extends>
   <header location="global">levelgraphicsview.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>levelPlistTableWidget</sender>
   <signal>itemChanged(QTableWidgetItem*)</signal>
   <receiver>MainWindow</receiver>
   <slot>levelPlistChanged(QTableWidgetItem*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>508</x>
     <y>226</y>
    </hint>
    <hint type="destinationlabel">
     <x>558</x>
     <y>-7</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>objectSelectorComboBox</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>updateObjectTable(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>661</x>
     <y>84</y>
    </hint>
    <hint type="destinationlabel">
     <x>569</x>
     <y>42</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>objectsTableWidget</sender>
   <signal>itemChanged(QTableWidgetItem*)</signal>
   <receiver>MainWindow</receiver>
   <slot>objectChanged(QTableWidgetItem*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>510</x>
     <y>270</y>
    </hint>
    <hint type="destinationlabel">
     <x>552</x>
     <y>37</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_4</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>addPropertyClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>512</x>
     <y>413</y>
    </hint>
    <hint type="destinationlabel">
     <x>790</x>
     <y>248</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_5</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>deletePropertyClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>512</x>
     <y>442</y>
    </hint>
    <hint type="destinationlabel">
     <x>793</x>
     <y>301</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_3</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>copyObjectClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>512</x>
     <y>487</y>
    </hint>
    <hint type="destinationlabel">
     <x>791</x>
     <y>403</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>newObjectClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>512</x>
     <y>516</y>
    </hint>
    <hint type="destinationlabel">
     <x>793</x>
     <y>470</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>deleteObjectClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>512</x>
     <y>545</y>
    </hint>
    <hint type="destinationlabel">
     <x>794</x>
     <y>504</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_6</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>addLevelPropertyClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>661</x>
     <y>515</y>
    </hint>
    <hint type="destinationlabel">
     <x>793</x>
     <y>436</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_7</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>deleteLevelPropertyClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>537</x>
     <y>543</y>
    </hint>
    <hint type="destinationlabel">
     <x>792</x>
     <y>533</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOpen_Level</sender>
   <signal>activated()</signal>
   <receiver>MainWindow</receiver>
   <slot>loadFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave_As</sender>
   <signal>activated()</signal>
   <receiver>MainWindow</receiver>
   <slot>saveLevelPlistAs()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave</sender>
   <signal>activated()</signal>
   <receiver>MainWindow</receiver>
   <slot>saveLevelPlist()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionNew_Level</sender>
   <signal>activated()</signal>
   <receiver>MainWindow</receiver>
   <slot>newLevel()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionExit</sender>
   <signal>activated()</signal>
   <receiver>MainWindow</receiver>
   <slot>quit()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>levelPlistChanged(QTableWidgetItem*)</slot>
  <slot>updateObjectTable(int)</slot>
  <slot>objectChanged(QTableWidgetItem*)</slot>
  <slot>addPropertyClicked()</slot>
  <slot>deletePropertyClicked()</slot>
  <slot>copyObjectClicked()</slot>
  <slot>newObjectClicked()</slot>
  <slot>deleteObjectClicked()</slot>
  <slot>addLevelPropertyClicked()</slot>
  <slot>deleteLevelPropertyClicked()</slot>
  <slot>loadFile()</slot>
  <slot>saveLevelPlist()</slot>
  <slot>saveLevelPlistAs()</slot>
  <slot>newLevel()</slot>
  <slot>quit()</slot>
 </slots>
</ui>
